File: C-Abi-Bridge-Aot/Tests/CAbiBridgeAot.Tests/TestCrypto/TestCryptoPqc/TestCryptoPqcSignature/TestCryptoPqcSignatureStateless/TestCryptoPqcSignatureMlDsa/Classes/MlDsaSignerInfo.cs

 

namespace michele.natale.Tests;


internal class MlDsaSignerInfo
{
  public byte[] SignerID
  {
    get; set; //Guid
  } = [];

  public byte[] Signature
  {
    get; set; //Sign + mldsa-algo
  } = [];

  public byte PqcSignAlgo //mldsa
  {
    get; set;
  } 

  public byte PqcSignAlgoParam // mldsa-algo
  {
    get; set;
  } 

  public byte[] PublicKey
  {
    get; set;
  } = [];

  public byte[] SignerName //Utf8
  {
    get; set;
  } = [];

  public byte[] ProjectName //Utf8
  {
    get; set;
  } = [];
}

File: C-Abi-Bridge-Aot/Tests/CAbiBridge.Aot.Vb.Tests/TestCrypto/TestCryptoPqc/TestCryptoPqcSignature/TestCryptoPqcSignatureStateless/TestCryptoPqcSignatureMlDsa/Classes/MlDsaSignerInfo.vb
Option Strict On
Option Explicit On


Namespace michele.natale.Tests
  Friend Class MlDsaSignerInfo
    Public Property SignerID As Byte() = Array.Empty(Of Byte)     'Guid
    Public Property Signature As Byte() = Array.Empty(Of Byte)    'Sign + mldsa-algo

    Public Property PqcSignAlgo As Byte                           'Here mldsa
    Public Property PqcSignAlgoParam As Byte                      'mldsa-algo

    Public Property PublicKey As Byte() = Array.Empty(Of Byte)
    Public Property SignerName As Byte() = Array.Empty(Of Byte)   'Utf8
    Public Property ProjectName As Byte() = Array.Empty(Of Byte)  'Utf8
  End Class
End Namespace

